src/OpenTelemetry/Batch.cs

```diff
@@ -46,7 +46,23 @@ internal Batch(CircularBuffer<T> circularBuffer, int maxSize)
 
             this.item = null;
             this.circularBuffer = circularBuffer ?? throw new ArgumentNullException(nameof(circularBuffer));
-            this.maxSize = maxSize;
+
+            long addedCount = circularBuffer.AddedCount;
+            long removedCount = circularBuffer.RemovedCount;
+            this.targetCount = removedCount + Math.Min(maxSize, addedCount - removedCount);
```

Comment:
```suggestion
            this.targetCount = removedCount + Math.Min(maxSize, circularBuffer.Count);
```

Reply:
I'll remove the locals and make it:
```chsharp
this.targetCount = circularBuffer.RemovedCount + Math.Min(maxSize, circularBuffer.Count);
```
The reason I had the locals was to snapshot the values in case they changed while it was executing but really we're the only reader so it should be fine (RemovedCount shouldn't change, AddedCount changing is fine).

Sound good?

Reply:
Yeah. I don't strong opinion on this (and I've already signed off) so up to you 😄